utils/uploads_used.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from datetime import datetime, timedelta
+
+from django.db.models import Q
+from django.utils import timezone
+
+from plan.constants import TrialStatus
+from reports.models import ReportSession
+from utils.config import get_config
+
+log = logging.getLogger(__name__)
+# default 6 hours for now
+cache_time = get_config("setup", "upload_usage_cache_time", 21600)
+
+
+def get_uploads_used(redis, plan_service, limit, owner):
+    cache_key = f"monthly_upload_usage_{owner.ownerid}"
```

Comment:
[nit] make the cache key some global lambda function in the file level. Reduces the number of places you need to update in case the key needs to change.

```python
CACHE_KEY = lambda ownerid: f"monthly_upload_usage_{ownerid}"


assert CACHE_KEY(10) == "monthly_upload_usage_10"
```